refactor: timeline manager renames

midzelis opened the pull request:
See comments in each commit for details of each rename, one step at a time.

midzelis committed:
refactor: timeline manager renames

midzelis committed:
refactor(web): improve timeline manager naming consistency

- Rename AddContext → GroupInsertionCache for clearer purpose
- Rename TimelineDay → DayGroup for better clarity
- Rename TimelineMonth → MonthGroup for better clarity
- Replace all "bucket" references with "monthGroup" terminology
- Update all component props, method names, and variable references
- Maintain consistent naming patterns across TypeScript and Svelte files

midzelis committed:
refactor(web): rename buckets to months in timeline manager

- Rename TimelineManager.buckets property to months
- Update all store.buckets references to store.months
- Use 'month' shorthand for monthGroup arguments (not method names)
- Update component templates and test files for consistency
- Maintain API-related 'bucket' terminology (bucketHeight, getTimeBucket)

midzelis committed:
refactor(web): rename assetStore to timelineManager and update types

- Rename assetStore variables to timelineManager in all .svelte files
- Update parameter names in actions.ts and asset-utils.ts functions
- Rename AssetStoreLayoutOptions to TimelineManagerLayoutOptions
- Rename AssetStoreOptions to TimelineManagerOptions
- Move assets-store.spec.ts to timeline-manager.spec.ts

midzelis committed:
refactor(web): rename intersectingAssets to viewerAssets and fix property references

- Rename intersectingAssets to viewerAssets in DayGroup and MonthGroup classes
- Update arrow function parameters to use viewerAsset/viewAsset shorthand
- Rename topIntersectingBucket to topIntersectingMonthGroup
- Fix dateGroups references to dayGroups in asset-utils.ts and album page
- Update template loops and variable names in Svelte components

midzelis committed:
refactor(web): rename #initializeTimeBuckets to #initializeMonthGroups and bucketDateFormatted to monthGroupTitle

midzelis committed:
refactor(web): rename monthGroupHeight to height

midzelis committed:
refactor(web): rename bucketCount to assetsCount, bucketsIterator to monthGroupIterator, and related properties

midzelis committed:
refactor(web): rename count to assetCount in TimelineManager

midzelis committed:
refactor(web): rename LiteBucket to ScrubberMonth and update scrubber variables

- Rename LiteBucket type to ScrubberMonth
- Rename bucketDateFormattted to title in ScrubberMonth type
- Rename bucketPercentY to monthGroupPercentY in scrubber component
- Rename scrubBucket to scrubberMonth and scrubBucketPercent to scrubberMonthPercent

midzelis committed:
fix remaining refs to bucket

midzelis committed:
reset submodule to correct commit

midzelis committed:
reset submodule to correct commit

midzelis committed:
refactor(web): extract TimelineManager internals into separate modules

- Move search-related functions to internal/search-support.svelte.ts
- Extract websocket event handling into WebsocketSupport class
- Move utility functions (updateObject, isMismatched) to internal/utils.svelte.ts
- Update imports in tests to use new module structure

midzelis committed:
refactor(web): extract intersection logic from TimelineManager

- Create intersection-support.svelte.ts with updateIntersection and calculateIntersecting functions
- Remove private intersection methods from TimelineManager
- Export findMonthGroupForAsset from search-support for reuse
- Update TimelineManager to use the extracted intersection functions

midzelis committed:
refactor(web): rename a few methods in intersecting

midzelis committed:
refactor(web): rename a few methods in intersecting

midzelis committed:
refactor(web): extract layout logic from TimelineManager

- Create layout-support.svelte.ts with updateGeometry and layoutMonthGroup functions
- Remove private layout methods from TimelineManager
- Update TimelineManager to use the extracted layout functions
- Remove unused UpdateGeometryOptions import

midzelis committed:
refactor(web): extract asset operations from TimelineManager

- Create operations-support.svelte.ts with addAssetsToMonthGroups and runAssetOperation functions
- Remove private asset operation methods from TimelineManager
- Update TimelineManager to use extracted operation functions with proper AssetOrder handling
- Rename getMonthGroupIndexByAssetId to getMonthGroupByAssetId for consistency
- Move utility functions from utils.svelte.ts to internal/utils.svelte.ts
- Fix method name references in asset-grid.svelte and tests

midzelis committed:
refactor(web): extract loading logic from TimelineManager

- Create load-support.svelte.ts with loadFromTimeBuckets function
- Extract time bucket loading, album asset handling, and error logging
- Simplify TimelineManager's loadMonthGroup method to use extracted function

midzelis committed:
refresh timeline after archive keyboard shortcut

midzelis committed:
remove debugger

midzelis committed:
rename

danieldietzler commented on web/src/lib/managers/timeline-manager/timeline-manager.svelte.ts:
```diff
@@ -0,0 +1,547 @@
+  }
+
+  addAssets(assets: TimelineAsset[]) {
+    const assetsToUpdate: TimelineAsset[] = [];
+
+    for (const asset of assets) {
+      if (this.isExcluded(asset)) {
+        continue;
+      }
+      assetsToUpdate.push(asset);
+    }
```
Could arguably also be a `const assetsToUpdate = assets.filter((asset) => this.isExcluded(asset))`

danieldietzler commented on web/src/lib/managers/timeline-manager/timeline-manager.svelte.ts:
```diff
@@ -0,0 +1,547 @@
+    if (asset) {
+      return asset;
+    }
+    for await (const asset of this.assetsIterator({ startMonthGroup: monthGroup })) {
+      return asset;
+    }
+  }
```
This could also just return `assetsIterator.next` I think

midzelis commented on web/src/lib/managers/timeline-manager/internal/layout-support.svelte.ts:
```diff
@@ -0,0 +1,73 @@
+    if (dayGroupCol > 0) {
+      rowSpaceRemaining[dayGroupRow] -= timelineManager.gap;
+    }
+    if (rowSpaceRemaining[dayGroupRow] >= 0) {
+      assetGroup.row = dayGroupRow;
+      assetGroup.col = dayGroupCol;
+      assetGroup.left = cummulativeWidth;
+      assetGroup.top = cummulativeHeight;
+
+      dayGroupCol++;
+
+      cummulativeWidth += assetGroup.width + timelineManager.gap;
+    } else {
+      cummulativeWidth = 0;
+      dayGroupRow++;
+      dayGroupCol = 0;
+      assetGroup.row = dayGroupRow;
+      assetGroup.col = dayGroupCol;
+      assetGroup.left = cummulativeWidth;
+
+      rowSpaceRemaining[dayGroupRow] -= assetGroup.width;
+      dayGroupCol++;
```
This algorithm might be confusing, so feedback to clarify is welcome. Basically, a month has several day groups. Usually, a day group fills the entire width of the timeline. However, some days with very little assets won't take up the full width. So, what we do is pack multiple days into a single row. The algorithm is very simple, start with the viewport width, subtract each day - if there's leftover space, continue putting additional days on the same row. A day-group is never split. Across 2 rows. A row either contains a single day, or multiple days. But if there is a day on a row, all the days on that row must fit on that row. If it there is a another day that fills the width after a bunch of days that don't, then that day gets it own row. 

Basically: 
  - dayGroupRow represents which row of day groups we're on
  - Multiple day groups can fit on the same row if there's space
  - Each day group stays intact (not split)


The initialization is performed line 33: 
```typescript
const rowSpaceRemaining: number[] = Array.from({ length: month.dayGroups.length });
rowSpaceRemaining.fill(timelineManager.viewportWidth, 0, month.dayGroups.length);
 ```

The array is sized based on the number of day groups (month.dayGroups.length), but dayGroupRow is used as an index for rows, not day groups. 

So the array rowSpaceRemaining with length month.dayGroups.length is actually oversized - it assumes worst case where each day group needs its own row, but in practice multiple day groups often fit on one row.


To your question: 
When we move to a new row (line 52), that row was already initialized at line 33 with the viewport width. The subtraction at line 58 is correct - it's tracking how much space remains on the new row after placing the current day group.

midzelis commented on web/src/lib/managers/timeline-manager/internal/load-support.svelte.ts:
```diff
@@ -0,0 +1,56 @@
+  }
+  const timeBucket = toISOYearMonthUTC(monthGroup.yearMonth);
+  const key = authManager.key;
+  const bucketResponse = await getTimeBucket(
```
This is the load function that is executed in a `CancelableTask` - the cancelableTask handles errors in a well-defined way. (Basically, transitioned into an error state, and calls the errorCallback)

midzelis committed:
Review comments - remove shadowed var

midzelis committed:
reduce indents - early return

midzelis committed:
review comment

midzelis commented on web/src/lib/managers/timeline-manager/internal/search-support.svelte.ts:
```diff
@@ -0,0 +1,149 @@
+async function getAssetByMonthOffset(timelineManager: TimelineManager, month: MonthGroup, direction: Direction) {
+  for (const targetMonth of timelineManager.monthGroupIterator({ startMonthGroup: month, direction })) {
+    if (targetMonth.yearMonth.month !== month.yearMonth.month) {
+      for await (const targetAsset of timelineManager.assetsIterator({ startMonthGroup: targetMonth, direction })) {
+        return targetAsset;
+      }
```
The full alternative would be 
```typescript
 const nextResult = await timelineManager.assetsIterator({ startMonthGroup: targetMonth, direction }).next();
 return nextResult?.value ?? undefined;
```
because next doesn't return the actual result, but rather a wrapper. I find the `for await` a little bit nicer - thoughts?

danieldietzler replied:
I think I prefer that tbh, though I'm not sure why you need `?? undefined`? Will nextResult be null?

midzelis committed:
refactor: simplify asset filtering in addAssets method

Replace for loop with filter operation for better readability

midzelis committed:
Merge remote-tracking branch 'origin/main' into refactor/rename

alextran1502 committed:
merge main

alextran1502 committed:
fix: bad merge

midzelis committed:
refactor(web): simplify timeline layout algorithm

- Replace rowSpaceRemaining array with direct cumulative width tracking
- Invert logic from tracking remaining space to tracking used space
- Fix spelling: cummulative to cumulative
- Rename lastRowHeight to currentRowHeight for clarity
- Remove confusing lastRow variable and simplify final height calculation
- Add explanatory comments for clarity
- Rename loop variable assetGroup to dayGroup for consistency

midzelis committed:
Merge branch 'refactor/rename' of github.com:immich-app/immich into refactor/rename

midzelis committed:
Merge remote-tracking branch 'origin/main' into refactor/rename

midzelis committed:
simplify assetsIterator usage

midzelis committed:
merge/lint
